fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    const cloneList = [...this.state.todos];
+    newTodo.todoId = (cloneList[cloneList.length - 1].todoId + 1);
+    cloneList.push(newTodo);
```

Comment:
You don't want to add the data from the form directly to state here. It doesn't have an `id` yet, the id is generated by the database, also the request hasn't completed yet so you are assuming the request will go okay when it could still fail.